pkg/validation/internal/multiarch.go

```diff
@@ -0,0 +1,513 @@
+//
+// Note: On this check, we aggregate the archetype(s) and OS(s) provided via the labels and those which are found by checking the images so that, we can check:
+//
+// - If your CSV is missing labels
+//
+// - If your Operator bundle specifies 1images which does not supports all archetypes found (probably supported by your project)
+//
+var MultipleArchitecturesValidator interfaces.Validator = interfaces.ValidatorFunc(multipleArchitecturesValidate)
+
+// ContainerToolsKey defines the key which can be used by its consumers
+// to inform where their container tool that should be used to inspect the image
+const ContainerToolsKey = "container-tools"
+
+// operatorFrameworkArchLabel defines the label used to the store the supported Arch on CSV
+const operatorFrameworkArchLabel = "operatorframework.io/arch."
+
+// operatorFrameworkSoLabel defines the label used to the store the supported So on CSV
+const operatorFrameworkSoLabel = "operatorframework.io/so."
+
+// MultiArchValidator store the data to perform the tests
+type MultiArchValidator struct {
+	// InfraArchLabels store the arch labels (i.e amd64, ppc64le) from
+	// operatorframework.io/arch.<GOARCH>: supported
+	InfraArchLabels []string
+	// InfraSOLabel store the OS labels from
+	// operatorframework.io/so.<GOARCH>: supported
+	InfraSOLabels []string
+	// Images stores the images defined in the bundle with the platform.arch supported
+	Images map[string][]Platform
+	// AllArchFound contains a map of the arch types found
+	AllArchFound map[string]string
+	// AllOsFound contains a map of the so(s) found
+	AllOsFound map[string]string
+	// Store the bundle load
+	Bundle *manifests.Bundle
+	// ContainerTool defines the container tool which will be used to inspect the images
+	ContainerTool string
+	// Warns stores the errors faced by the validator to return the warnings
+	Warns []error
+}
+
+// ManifestInspect store the data obtained by running container-tool manifest inspect <IMAGE>
+type ManifestInspect struct {
+	ManifestData []ManifestData `json:"manifests"`
+}
+
+// ManifestData store the platforms
+type ManifestData struct {
+	Platform Platform `json:"platform"`
+}
+
+// Platform store the Architecture and OS supported by the image
+type Platform struct {
+	Architecture string `json:"architecture"`
+	OS           string `json:"os"`
+}
+
+func multipleArchitecturesValidate(objs ...interface{}) (results []errors.ManifestResult) {
+	// Obtain the k8s version if informed via the objects an optional
+	var containerTool = ""
+	for _, obj := range objs {
+		switch obj.(type) {
+		case map[string]string:
+			containerTool = obj.(map[string]string)[ContainerToolsKey]
+			if len(containerTool) > 0 {
+				break
+			}
+		}
+	}
+
+	for _, obj := range objs {
+		switch v := obj.(type) {
+		case *manifests.Bundle:
+			results = append(results, validateMultiArchWith(v, containerTool))
+		}
```

Comment:
Similar comment to above:
```go
if bundle, ok := obj.(*manifests.Bundle); ok {
    // append bundle to results
}
```

Reply:
same here. If we have other types? 
I think I would prefer to stick with the current implementation which indeed is == across all validators.